pkg/scheduler/framework/interface.go

```diff
@@ -249,21 +252,22 @@ func (s *Status) Equal(x *Status) bool {
 	if s.code != x.code {
 		return false
 	}
-	if s.code == Error {
+	if s.code == Error || s.err != nil {
 		return cmp.Equal(s.err, x.err, cmpopts.EquateErrors())
 	}
-	return cmp.Equal(s.reasons, x.reasons)
+	return cmp.Equal(s.Reasons(), x.Reasons())
 }
 
 // NewStatus makes a Status out of the given arguments and returns its pointer.
 func NewStatus(code Code, reasons ...string) *Status {
 	s := &Status{
-		code:    code,
-		reasons: reasons,
+		code: code,
 	}
 	if code == Error {
-		s.err = errors.New(s.Message())
+		s.err = errors.New(strings.Join(reasons, ", "))
+		return s
 	}
```

Comment:
Just remove this.

We don't need to parse the error at this point. `AsError` does that when necessary.

Reply:
consider this scenario:
```
err := errors.New("error")
x := AsStatus(err)
y := NewStatus(Error, "error")
```

Currently, x and y are considered not equal, cause x has `s.err!=nil` and y has `s.err !=nil ` since we don't constuct error here, is this reasonable?

Reply:
Also, I'm confused by this unit test case:
```
name: "error statuses with same message should not be equal",
x:    NewStatus(Error, "error"),
y:    NewStatus(Error, "error"),
want: false,
```

It seems that `s.Eaual()` will return false when it compares these two errors

Reply:
> Currently, x and y are considered not equal, cause x has s.err!=nil and y has s.err ==nil since we don't constuct error in NewStatus, is this reasonable?

That is correct. They are different errors that happen to have the same message.

```
x:    NewStatus(Error, "error"),
y:    NewStatus(Error, "error"),
```

I think we could consider these ones equal, but not this:

```
x:    AsStatus(errors.New("error")),
y:    AsStatus(errors.New("error")),
```

Reply:
Got you, when two status both have errors, we check if two errors are the same one. Otherwise we juse check the code and reason.

You can review changes in `pkg/scheduler/framework/interface.go`  first, I'll update unit tests next.